pkg/apis/options/legacy_options.go

```diff
@@ -555,6 +556,7 @@ func legacyProviderFlagSet() *pflag.FlagSet {
 	flagSet.String("client-id", "", "the OAuth Client ID: ie: \"123456.apps.googleusercontent.com\"")
 	flagSet.String("client-secret", "", "the OAuth Client Secret")
 	flagSet.String("client-secret-file", "", "the file with OAuth Client Secret")
+	flagSet.Bool("client-secret-allow-empty", false, "allow omitting the Client Secret (e.g. when using PKCE without a Client Secret)")
```

Comment:
Apologies for adding this to legacy options, I did have a look at adding this to provider, but seems like there isn't a providerFlagSet similar to cookieFlagSet to add this option to in the new way of defining the options. So I've added it here so that all these provider options could possibly be refactored to the new way together.